src/station_cmd.cpp

```diff
@@ -3669,11 +3681,15 @@ void StationMonthlyLoop()
 
 void ModifyStationRatingAround(TileIndex tile, Owner owner, int amount, uint radius)
 {
-	Station *st;
-
-	FOR_ALL_STATIONS(st) {
-		if (st->owner == owner &&
-				DistanceManhattan(tile, st->xy) <= radius) {
+	uint16 x1, y1, x2, y2;
+	x1 = (uint16)max<int>(0, TileX(tile) - radius);
+	x2 = (uint16)min<int>(TileX(tile) + radius, MapMaxX());
+	y1 = (uint16)max<int>(0, TileY(tile) - radius);
+	y2 = (uint16)min<int>(TileY(tile) + radius, MapMaxY());
```

Comment:
It has a different style than in `GetClosestDeletedStation`.
Also, there you use `MapSizeX`, here `MapMaxX`. If you can place a station at the edge of the map, `MapSizeX` is better.

Reply:
I think I intended to factor the search for stations within a tile radius into a function of its own.